pwiz_tools/Skyline/FileUI/PeptideSearch/EncyclopeDiaSearchDlg.cs

```diff
@@ -0,0 +1,611 @@
+            pageControl.Location = new Point(border, header);
+            pageControl.Width = tabPage.Width - border * 2;
+            pageControl.Height = tabPage.Height - header - border;
+            tabPage.Controls.Add(pageControl);
+        }
+
+        private readonly LibraryManager _libraryManager;
+        private readonly Stack<SrmDocument> _documents;
+        public SrmDocument Document
+        {
+            get
+            {
+                if (_documents.Count == 0)
+                    return null;
+                return _documents.Peek();
+            }
+        }
+
+        public string DocumentFilePath
+        {
+            get => SkylineWindow.DocumentFilePath;
+            set => SkylineWindow.DocumentFilePath = value;
+        }
+
+        public bool SetDocument(SrmDocument docNew, SrmDocument docOriginal)
+        {
+            if (!ReferenceEquals(Document, docOriginal))
+                return false;
+
+            _documents.Push(docNew);
+            return true;
+        }
+
+        #region IModifyDocumentContainer
+        public void Listen(EventHandler<DocumentChangedEventArgs> listener)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Unlisten(EventHandler<DocumentChangedEventArgs> listener)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool IsClosing { get { throw new NotImplementedException(); } }
+        public IEnumerable<BackgroundLoader> BackgroundLoaders { get { throw new NotImplementedException(); } }
+        public void AddBackgroundLoader(BackgroundLoader loader)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void RemoveBackgroundLoader(BackgroundLoader loader)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void ModifyDocumentNoUndo(Func<SrmDocument, SrmDocument> act)
+        {
+            ModifyDocument(null, act, null);
+        }
+
+        public void ModifyDocument(string description, Func<SrmDocument, SrmDocument> act, Func<SrmDocumentPair, AuditLogEntry> logFunc)
+        {
+            var docNew = act(Document);
+            if (ReferenceEquals(docNew, Document))
+                return;
+
+            SetDocument(docNew, Document);
+        }
+        #endregion
+
+        public Pages CurrentPage => (Pages) wizardPages.SelectedIndex;
+
+        public void NextPage()
+        {
+            if (!File.Exists(ImportFastaControl.FastaFile))
+            {
+                MessageDlg.Show(this, Resources.EncyclopeDiaSearchDlg_NextPage_A_FASTA_file_is_required_for_an_EncyclopeDia_Prosit_search_);
+                return;
+            }
+
+            btnNext.Text = Resources.BuildLibraryDlg_btnPrevious_Click__Next__;
+
+            if (CurrentPage == Pages.run_page)
+            {
+                ImportEncyclopediaLibrary();
+                return;
```

Comment:
@nickshulman @brendanx67 Running EncyclopeDia doesn't actually change the document until the import peptide search wizard adds the libraries created in the search. But it would make sense to me to give EncyclopeDia its own entry though: what is the proper way to add an audit log entry when the document doesn't change?

Reply:
There is really no such thing as this. Again, I think we probably want to expand our understanding of auxiliary files that might get included with a File > Share. So, probably you will want to modify the document with something that points at an EncyclopeDIA product like a parameters file or something that would allow you to re-run the analysis from scratch.

Reply:
Yeah. Maybe it is worth having a schema element for all searches with name-value pairs for parameters to store how the search was run to make it easier to re-run